src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlConverter.cs

```diff
@@ -1082,45 +1086,63 @@ public static int ToChars(DateTime value, byte[] chars, int offset)
             return offset - offsetMin;
         }
 
-        public static bool IsWhitespace(string s)
+        public static bool IsWhitespace(ReadOnlySpan<char> chars) =>
+            chars.IndexOfAnyExcept(s_whitespaceChars) < 0;
+
+        public static bool IsWhitespace(ReadOnlySpan<byte> bytes) =>
+            bytes.IndexOfAnyExcept(s_whitespaceBytes) < 0;
+
+        public static bool IsWhitespace(char ch) =>
+            ch is <= ' ' and (' ' or '\t' or '\r' or '\n');
+
+        public static int StripWhitespace(Span<char> chars)
         {
-            for (int i = 0; i < s.Length; i++)
+            int count = chars.IndexOfAny(s_whitespaceChars);
+            if (count < 0)
             {
-                if (!IsWhitespace(s[i]))
-                    return false;
+                return chars.Length;
             }
-            return true;
-        }
 
-        public static bool IsWhitespace(char ch)
-        {
-            return (ch <= ' ' && (ch == ' ' || ch == '\t' || ch == '\r' || ch == '\n'));
+            for (int i = count; i < chars.Length; i++)
+            {
+                char c = chars[i];
+                if (!IsWhitespace(c))
+                {
+                    chars[count++] = c;
+                }
+            }
+
+            return count;
         }
 
         public static string StripWhitespace(string s)
         {
+            int indexOfWhitespace = s.AsSpan().IndexOfAny(s_whitespaceChars);
+            if (indexOfWhitespace < 0)
+            {
+                return s;
+            }
+
             int count = s.Length;
```

Comment:
should we just initialize the count with `s.Length - 1` and start the loop from `indexOfWhitespace + 1`?

Reply:
Sure, we can do that.